src/Slim/Middleware/Session.php

```diff
@@ -79,6 +84,15 @@ protected function startSession()
     {
         $settings = $this->settings;
         $name = $settings['name'];
+        $handler = $settings['handler'];
+
+        if ($handler) {
+            if ($handler instanceof Handler) {
+                $this->registHandler($handler);
```

Comment:
I think this should be at `__construct`, correct me if I'm wrong.

Reply:
No, It should called when session middleware invoked to avoid user call ```session_set_save_handler(...)``` after ```__construct(...)```

Reply:
I'll take a look at it, hope the official docs tell where to put that call.